client/web/src/insights/pages/search-insight-creation/components/form-series-input/FormSeriesInput.tsx

```diff
@@ -0,0 +1,192 @@
+            if (nameField.meta.error) {
+                event.preventDefault()
+                return nameReference.current?.focus()
+            }
+
+            if (queryField.meta.error) {
+                event.preventDefault()
+                return queryReference.current?.focus()
+            }
+
+            // handleSubmit work with form element and use form event
+            // but we can't have sub forms for the sake of semantics.
+            // if this case synthetic event of totally comparable with event
+            // from submit button.
+            await handleSubmit((event as unknown) as React.SyntheticEvent<HTMLFormElement>)
+        },
+        [handleSubmit, nameField.meta.error, queryField.meta.error]
+    )
+
+    return (
+        <div className={classnames(styles.formSeriesInput, className)}>
+            <InputField
+                title="Name"
+                placeholder="ex. Function component"
+                description="Name shown in the legend and tooltip"
+                valid={(hasNameControlledValue || nameField.meta.touched) && nameField.meta.valid}
+                error={nameField.meta.touched && nameField.meta.error}
+                className={styles.formSeriesInputField}
+                {...nameField.input}
+                ref={nameReference}
+            />
+
+            <InputField
+                title="Query"
+                placeholder="ex. spatternType:regexp const\\s\\w+:\\s(React\\.)?FunctionComponent"
+                description="Do not include the repo: filter as it will be added automatically for the current repository"
+                valid={(hasQueryControlledValue || queryField.meta.touched) && queryField.meta.valid}
+                error={queryField.meta.touched && queryField.meta.error}
+                className={styles.formSeriesInputField}
+                {...queryField.input}
+                ref={queryReference}
+            />
+
+            <FormColorInput
+                name="series color group"
+                title="Color"
+                className={classnames(styles.formSeriesInputField, styles.formSeriesInputColor)}
+                value={colorField.input.value}
+                onChange={colorField.input.onChange}
+            />
+
+            <div>
+                <button
+                    type="submit"
+                    onClick={handleSubmitButton}
+                    className={classnames(styles.formSeriesInputButton, 'button')}
+                >
+                    Done
+                </button>
+
+                {cancel && (
+                    <button
+                        type="button"
+                        onClick={onCancel}
+                        className={classnames(
+                            styles.formSeriesInputButton,
+                            styles.formSeriesInputButtonCancel,
+                            'button'
+                        )}
+                    >
+                        Cancel
```

Comment:
I think "Discard" is a bit more fitting as this is more of a draft you're throwing away, as opposed to an ongoing operation you want to abort/cancel.
```suggestion
                        Discard
```
@quinnkeast do we have content guidelines on this?

Reply:
I totally understand the argument here... however, "Cancel" is still a logical opposition to the action being noped-out of, which in this case, is "**Create** / **Cancel**," and in the future, "**Edit** or **Save changes** / Cancel". I think consistency in the action label here works well. 

(Whatever changes we do here, would logically then apply to code monitoring and search contexts in the future.)